src/message/attachment.rs

```diff
@@ -0,0 +1,137 @@
+use crate::message::{
+    header::{self, Header},
+    IntoBody, SinglePart,
+};
+use mime::Mime;
+
+#[derive(Clone, Copy)]
+enum Disposition {
+    Attachment,
+    Inline,
+}
+
+/// `SinglePart` builder for attachments
+///
+/// Allows building attachment parts easily.
+pub struct Attachment {
+    filename: Option<String>,
+    content_disposition: Disposition,
+    content_type: Option<Mime>,
+    content_id: Option<String>,
+}
+
+impl Default for Attachment {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Attachment {
+    /// Creates a new attachment
+    pub fn new() -> Self {
+        Self {
+            filename: None,
+            content_disposition: Disposition::Attachment,
+            content_type: None,
+            content_id: None,
+        }
+    }
+
+    /// Creates a new inline attachment
+    pub fn new_inline() -> Self {
+        Self {
+            filename: None,
+            content_disposition: Disposition::Inline,
+            content_type: None,
+            content_id: None,
+        }
+    }
+
+    /// Specify a MIME type for the attachment
+    pub fn content_type(mut self, content_type: Mime) -> Self {
```

Comment:
This could use `ContentType`, instead of exposing the mime crate into the public API